course6/toyapp-adam/math/vector.h

```diff
@@ -1,113 +1,172 @@
 #pragma once
 
+#include <cassert>
+#include <array>
 #include <cmath>
+#include <limits>
 
 namespace Math {
 
-struct Vector2D
+template<typename Type, unsigned Dimensions, typename FloatType = float>
```

Comment:
You don't need the ```FloatType``` to be the property of the entire class. You can use it as a template parameter on the methods that need it, so that you can use different floating-point types with the same Vector

Reply:
You're right, this actually crossed my mind after I created the PR